naggie/dstask
# Add fish completions, update documentation accordingly

Closes #159 

The fish shell's [completions docs](https://fishshell.com/docs/current/completions.html) expect a space-separated list of completions. Piping `dtask _completions` into the fish utility [`string-collect`](https://fishshell.com/docs/current/cmds/string.html#collect-subcommand) appears to do exactly what is needed to setup the completions.

- Adds a file "completions/completions.fish"

**Testing**

- Install fish
- From your bash shell, run `fish` to enter a fish shell
- Checkout my branch and run `make dist/dstask`
- Run `dist/dstask fish-completion | source`
- Start typing `dist/dstask show-` and see now that the completions work in Fish

**Reviewer guidance**

Have I missed anything? I basically just grepped for "bash" and then copy-pasted the seemingly relevant lines changed "bash" to "fish".

I've also updated the documentation to show multiple examples -- one each for bash, zsh and fish. I felt this would be particularly helpful for fish users as the `<()` operation doesn't work in fish:

```fish
fish: Invalid redirection target:
source <(dist/dstask _completions)
```

Happy to tweak it as necessary.
